node_modules/oae-authentication/lib/api.js

```diff
@@ -674,6 +683,118 @@ var getUserIdFromLoginId = module.exports.getUserIdFromLoginId = function(tenant
     _getUserIdFromLoginId(loginId, callback);
 };
 
+
+/**
+ * Get the secret for an existing user's resetpassword request
+ *
+ * @param  {Context}    ctx               Standard context object containing the current user and the current tenant
+ * @param  {String}     userName		  The user's own unique username 
+ * @param  {Function}   callback          Standard callback function
+ * @param  {Object}     callback.err      An error that occurred, if any
+ */
+var getResetPasswordSecret =module.exports.getResetPasswordSecret = function(ctx, tenantAlias, username, callback){
+	
+	// associate the userID with loginID for later operations
+    tenantAlias = tenantAlias || ctx.tenant().alias;
+
+    // Create the loginid object first, so it can be passed into the validation
+    var loginId = new LoginId(tenantAlias, AuthenticationConstants.providers.LOCAL, username);
+	
+	// generate a secret token
+	var secret = crypto.randomBytes(16).toString('hex');
+	
+	// check if we can get an existing userID using the userName the user provided
+	_getUserIdFromLoginId(loginId, function(err, userId) {
```

Comment:
We have the convention to check whether an error occurred first and return from the function if one did. It saves a bit on indentation.
 i.e.,
```javascript
_getUserIdFromLoginId(loginId, function(err, userId) {
    if (err) {
        return callback(err);
    } else if (!userId) {
        callback({'code': 400, 'msg': 'This login id does not exist and is not associated to any user'});
    }

    // More logic here..
});
```


That said, I *think* the `_getUserIdFromLoginId` will return a 404 if the userId could not be found so there might be no need for the `else if` clause.